sklearn/tests/test_multiclass.py

```diff
@@ -260,8 +260,8 @@ def test_ovr_binary():
     def conduct_test(base_clf, test_predict_proba=False):
         clf = OneVsRestClassifier(base_clf).fit(X, y)
         assert_equal(set(clf.classes_), classes)
-        y_pred = clf.predict(np.array([[0, 0, 4]]))[0]
-        assert_equal(set(y_pred), set("eggs"))
+        y_pred = clf.predict(np.array([[0, 0, 4]]))
+        assert_equal(set(y_pred), {"eggs"})
```

Comment:
@qinhanmin2014  So `set(a)` will iterate on `a` and create a set with items of a. 

In the original code we  iterated over the first element of `y_pred`, which was a string, and so the result was `set("eggs") == {'g', 'e', 's'}`. The same thing was done on the right. It worked but I'm pretty sure that was not intentional.

Here, I changed the code a bit following your comment. On the left we iterate on `y_pred` that only has one element `"egg"` and on the right, we create the same set with one element.